src/wallet/spend.cpp

```diff
@@ -692,7 +692,9 @@ util::Result<SelectionResult> ChooseSelectionResult(interfaces::Chain& chain, co
     };
 
     // Maximum allowed weight for selected coins.
-    int max_selection_weight = MAX_STANDARD_TX_WEIGHT - (coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR);
+    int max_transaction_weight = coin_selection_params.m_max_tx_weight.value_or(MAX_STANDARD_TX_WEIGHT);
+    int tx_weight_no_input = coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR;
+    int max_selection_weight = std::max((max_transaction_weight - tx_weight_no_input), 0);
```

Comment:
In d9ec6e7f28f0:

Instead of setting it to 0, could error out early if `max_selection_weight <= 0` (a test for this would be nice).

Reply:
Thanks fixed.

Reply:
Also added a test for this